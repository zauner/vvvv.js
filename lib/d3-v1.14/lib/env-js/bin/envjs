#!/bin/sh
# 	Usage: envjs <platorm> file.js [file2.js ...]
###########################################################################
ENVJS_PLATFORM='node'

# 	first arguments ($1) 
#		may be a platform or is a file and the default platorm is used
if [ -n "$1" ]; then 
    if [ -n "$1" ]; then ENVJS_PLATFORM="$1"; shift; fi
fi

#   Run envjs with the given platform
###########################################################################
case "$ENVJS_PLATFORM" in

    "node")
        node envjs/node.js $@
        ;;
    "rhino")
        java -Xmx512M -jar rhino/js.jar -opt -1 envjs/rhino.js $@
        ;;
    "rhino-debug")
        java -cp rhino/js.jar org.mozilla.javascript.tools.debugger.Main envjs/rhino.js $@
        ;;
    "spyd")
        python envjs/spydermonkey.py envjs/spydermonkey.js $@
        ;;
    "rubyracer")
        ruby -rrubygems envjs/rubyracer.rb envjs/rubyracer.js $@
        ;;
    "johnson")
        ruby -rrubygems envjs/johnson.rb envjs/johnson.js $@
        ;;
	*)
		#	platform default means $1 was actually a file
        node envjs/node.js $ENVJS_PLATFORM $@
        ;;
esac

