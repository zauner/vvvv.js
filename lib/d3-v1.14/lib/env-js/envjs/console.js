
/**
 * @author envjs team
 * @Console
 */

var Envjs = Envjs || require('./platform/core').Envjs;

/*
 * Envjs console.1.3.pre03
 * Pure JavaScript Browser Environment
 * By John Resig <http://ejohn.org/> and the Envjs Team
 * Copyright 2008-2010 John Resig, under the MIT License
 */

//CLOSURE_START
(function(){




/**
 * @author envjs team
 * borrowed 99%-ish with love from firebug-lite
 */

//leaked globally on purpose;
try{
    console.log();
}catch(e){


function escapeHTML(value)
{
    return value;
}

function objectToString(object)
{
    try
    {
        return object+"";
    }
    catch (exc)
    {
        return null;
    }
}

// ********************************************************************************************

function appendText(object, html)
{
    html.push(escapeHTML(objectToString(object)));
}

function appendNull(object, html)
{
    html.push(escapeHTML(objectToString(object)));
}

function appendString(object, html)
{
    html.push(escapeHTML(objectToString(object)));
}

function appendInteger(object, html)
{
    html.push(escapeHTML(objectToString(object)));
}

function appendFloat(object, html)
{
    html.push(escapeHTML(objectToString(object)));
}

function appendFunction(object, html)
{
    var reName = /function ?(.*?)\(/;
    var m = reName.exec(objectToString(object));
    var name = m ? m[1] : "function";
    html.push(escapeHTML(name));
}

function appendObjectFormatted(object, html)
{
    var text = objectToString(object);
    var reObject = /\[object (.*?)\]/;

    var m = reObject.exec(text);
    html.push( m ? m[1] : text);
}

function appendSelector(object, html)
{

    html.push(escapeHTML(object.nodeName.toLowerCase()));
    if (object.id) {
        html.push(escapeHTML(object.id));
    }
    if (object.className) {
        html.push(escapeHTML(object.className));
    }
}

function appendNode(node, html)
{
    if (node.nodeType == 1)
    {
        html.push( node.nodeName.toLowerCase());

        for (var i = 0; i < node.attributes.length; ++i)
        {
            var attr = node.attributes[i];
            if (!attr.specified) {
                continue;
        }

            html.push( attr.nodeName.toLowerCase(),escapeHTML(attr.nodeValue));
        }

        if (node.firstChild)
        {
            for (var child = node.firstChild; child; child = child.nextSibling) {
                appendNode(child, html);
        }

            html.push( node.nodeName.toLowerCase());
        }
    }
    else if (node.nodeType === 3)
    {
        html.push(escapeHTML(node.nodeValue));
    }
}

function appendObject(object, html)
{
    try
    {
        if (object === undefined) {
            appendNull("undefined", html);
        } else if (object === null) {
            appendNull("null", html);
        } else if (typeof object == "string") {
            appendString(object, html);
    } else if (typeof object == "number") {
            appendInteger(object, html);
    } else if (typeof object == "function") {
            appendFunction(object, html);
        } else if (object.nodeType == 1) {
            appendSelector(object, html);
        } else if (typeof object == "object") {
            appendObjectFormatted(object, html);
        } else {
            appendText(object, html);
    }
    }
    catch (exc)
    {
    }
}


function parseFormat(format)
{
    var parts = [];

    var reg = /((^%|[^\\]%)(\d+)?(\.)([a-zA-Z]))|((^%|[^\\]%)([a-zA-Z]))/;
    var appenderMap = {s: appendText, d: appendInteger, i: appendInteger, f: appendFloat};

    for (var m = reg.exec(format); m; m = reg.exec(format))
    {
        var type = m[8] ? m[8] : m[5];
        var appender = type in appenderMap ? appenderMap[type] : appendObject;
        var precision = m[3] ? parseInt(m[3], 10) : (m[4] == "." ? -1 : 0);

        parts.push(format.substr(0, m[0][0] == "%" ? m.index : m.index+1));
        parts.push({appender: appender, precision: precision});

        format = format.substr(m.index+m[0].length);
    }

    parts.push(format);

    return parts;
}



function logFormatted(objects, className)
{
    var html = [],
        i= 0,
        object;

    var format = objects[0];
    var objIndex = 0;

    if (typeof(format) != "string")
    {
        format = "";
        objIndex = -1;
    }

    var parts = parseFormat(format);
    for (i = 0; i < parts.length; ++i)
    {
        var part = parts[i];
        if (part && typeof(part) == "object")
        {
            object = objects[++objIndex];
            part.appender(object, html);
        }
        else {
            appendText(part, html);
        }
    }

    for (i = objIndex+1; i < objects.length; ++i)
    {
        appendText(" ", html);

        object = objects[i];
        if (typeof(object) == "string") {
            appendText(object, html);
        } else {
            appendObject(object, html);
        }
    }

    Envjs.log(html.join(' '));
}


Console = function(module){
    var $level,
        $logger,
        $null = function(){};
    $logger = {
        log: function(level){
            logFormatted(arguments, "");
        },
        debug: function(level){
            logFormatted(arguments, "DEBUG");
        },
        info:  function(level){
            logFormatted(arguments, "INFO");
        },
        warn:  function(level){
            logFormatted(arguments, "WARN");
        },
        error:  function(level){
            logFormatted(arguments, "ERROR");
        },
        trace: function(){
            Envjs.trace();
        }
    };

    return $logger;
};

console = new Console();
exports.console = console;

}



/**
 * @author john resig & the envjs team
 * @uri http://www.envjs.com/
 * @copyright 2008-2010
 * @license MIT
 */
//CLOSURE_END
}());
