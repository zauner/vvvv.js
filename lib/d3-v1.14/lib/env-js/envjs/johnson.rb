require 'johnson'
require 'net/http'
require 'uri'
require 'thread'

include Config

def configure_context(context)
  context['global']    = context
  context['HTTPConnection'] = HTTPConnection.new
end

class HTTPConnection
  def go(connection, request, headers, data)
    headers.each{|key,value| request.add_field(key,value)}
    response, body = connection.request(request, data)
    respheaders = Hash.new
    response.each_header do |name, value|
      respheaders.store(name, value)
    end
    response['body'] = body
    [response, respheaders]
  end
end

global = Johnson::Runtime.new
configure_context(global)
envjs = ARGV[0]
global.load(envjs)


