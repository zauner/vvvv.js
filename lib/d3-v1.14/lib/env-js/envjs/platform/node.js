var __context__ = __this__ = global;

var Envjs = Envjs ||
	require('./core').Envjs;
	require('./../../local_settings');

Envjs.platform       = "Node";
Envjs.revision       = process.version;

/*process.on('uncaughtException', function (err) {
  	console.log('Envjs Caught exception: %s \n %s', err);
});*/

Envjs.argv = process.argv;
Envjs.argv.shift();
Envjs.argv.shift();//node is argv[0] but we want to start at argv[1]

Envjs.exit = function(){
    /*setTimeout(function () {
        if(!Envjs.timers.length){
            //console.log('no timers remaining %s', Envjs.timers.length);
            process.exit();
        }else{
            Envjs.exit();
        }
    }, 13);*/
};


/*
 * Envjs node-env.1.3.pre03
 * Pure JavaScript Browser Environment
 * By John Resig <http://ejohn.org/> and the Envjs Team
 * Copyright 2008-2010 John Resig, under the MIT License
 */

//CLOSURE_START
(function(){





/**
 * @author john resig
 */
// Helper method for extending one object with another.
function __extend__(a,b) {
    for ( var i in b ) {
        if(b.hasOwnProperty(i)){
            var g = b.__lookupGetter__(i), s = b.__lookupSetter__(i);
            if ( g || s ) {
                if ( g ) { a.__defineGetter__(i, g); }
                if ( s ) { a.__defineSetter__(i, s); }
            } else {
                a[i] = b[i];
            }
        }
    }
    return a;
}

var $print = require('sys').print;

Envjs.log = function(msg){
    console.log(msg+'\n\n');
};

Envjs.lineSource = function(e){
    return "(line ?)";
};

Envjs.readConsole = function(){
     return $stdin.gets();
};

Envjs.prompt = function(){
    $stdout.write(Envjs.CURSOR+' ');
    $stdout.flush;
};

//No REPL for
Envjs.repl = function(){
    //require('repl').start(Envjs.CURSOR+' ');
    console.log('Envjs REPL Not Available');
};
var Script = process.binding('evals').Script;

Envjs.eval = function(context, source, name, warming){
	if(context === global){
		return warming ?
			eval(source) :
			Script.runInThisContext(source, name);
	}else{
		return Script.runInNewContext(source, context, name);
	}
};
Envjs.wait = function(){ return; }

var $tick = function(){
    process.nextTick(function () {
        //console.log('node tick');
        Envjs.tick();
        $tick();
    });
};

Envjs.eventLoop = function(){
    //console.log('event loop');
    $tick();
};
/**
 * provides callback hook for when the system exits
 */
Envjs.onExit = function(callback){
    process.on('exit', callback);
};


/**
 * synchronizes thread modifications
 * @param {Function} fn
 */
 Envjs.sync = function(fn){
     //console.log('syncing js fn %s', fn);
     return fn;
 };

 Envjs.spawn = function(fn){
     return fn();
 };

/**
 * sleep thread for specified duration
 * @param {Object} milliseconds
 */
Envjs.sleep = function(milliseconds){
    return;
};


//Since we're running in v8 I guess we can safely assume
//java is not 'enabled'.  I'm sure this requires more thought
//than I've given it here
Envjs.javaEnabled 	 = false;

Envjs.homedir        = process.env["HOME"];
Envjs.tmpdir         = process.env["TMPDIR"];
Envjs.os_name        = process.platform;
Envjs.os_arch        = "os arch";
Envjs.os_version     = "os version";
Envjs.lang           = process.env["LANG"];


Envjs.gc = function(){ return; };

/**
 * Makes an object window-like by proxying object accessors
 * @param {Object} scope
 * @param {Object} parent
 */
Envjs.proxy = function(scope, parent) {
    try{
        if(scope.toString() == global){
            return __this__;
        }else{
            return scope;
        }
    }catch(e){
       console.log('failed to init standard objects %s %s \n%s', scope, parent, e);
    }

};var filesystem = require('fs');
/**
 * Get 'Current Working Directory'
 */
Envjs.getcwd = function() {
    return process.cwd();
}


/**
 * Used to write to a local file
 * @param {Object} text
 * @param {Object} url
 */
Envjs.writeToFile = function(text, url){
	if(/^file\:\/\//.test(url))
		url = url.substring(7,url.length);
    filesystem.writeFileSync(url, text, 'utf8');
};

/**
 * Used to write to a local file
 * @param {Object} text
 * @param {Object} suffix
 */
Envjs.writeToTempFile = function(text, suffix){
	var url =  Envjs.tmpdir+'envjs-'+(new Date().getTime())+'.'+suffix;
	if(/^file\:\/\//.test(url))
		url = url.substring(7,url.length);
	filesystem.writeFileSync(tmpfile, text, 'utf8');
    return tmpfile;
};


/**
 * Used to read the contents of a local file
 * @param {Object} url
 */
Envjs.readFromFile = function( url ){
	if(/^file\:\/\//.test(url))
		url = url.substring(7,url.length);
    return filesystem.readFileSync(url, 'utf8');
};


/**
 * Used to delete a local file
 * @param {Object} url
 */
Envjs.deleteFile = function(url){
	if(/^file\:\/\//.test(url))
		url = url.substring(7,url.length);
    filesystem.unlink(url);
};

/**
 * establishes connection and calls responsehandler
 * @param {Object} xhr
 * @param {Object} responseHandler
 * @param {Object} data
 */
Envjs.connection = function(xhr, responseHandler, data){
    var url = xhr.url,
        connection,
		request,
        binary = false,
        contentEncoding,
        responseXML = null,
		http = require('http'),
		urlparts = Envjs.urlsplit(url),
        i;


    if ( /^file\:/.test(url) ) {
        Envjs.localXHR(url, xhr, connection, data);
    } else {
	    //console.log('connecting to %s \n\t port(%s) host(%s) path(%s) query(%s)',
	    //    url, urlparts.port, urlparts.hostname, urlparts.path, urlparts.query);
		connection = http.createClient(urlparts.port||'80', urlparts.hostname);
		request = connection.request(
			xhr.method,
			urlparts.path+(urlparts.query?"?"+urlparts.query:''),
			__extend__(xhr.headers,{
				"Host": urlparts.hostname,
				//"Connection":"Keep-Alive"
				//"Accept-Encoding", 'gzip'
			})
		);
		xhr.statusText = "";

	    if(connection&&request){
			request.on('response', function (response) {
				//console.log('response begin');
				xhr.readyState = 3;
				response.on('end', function (chunk) {
					//console.log('connection complete');
		        	xhr.readyState = 4;
				    if(responseHandler){
				        //console.log('calling ajax response handler');
				        if(!xhr.async){
							responseHandler();
						}else{
							setTimeout(responseHandler, 1);
						}
				    }
					if(xhr.onreadystatechange){
						xhr.onreadystatechange();
					}
				});

				xhr.responseHeaders = __extend__({},response.headers);
                xhr.statusText = "OK";
			    xhr.status = response.statusCode;
				//console.log('response headers : %s', JSON.stringify(xhr.responseHeaders));
				contentEncoding = xhr.getResponseHeader('Content-Encoding') || "utf-8";
				response.setEncoding(contentEncoding);
	            //console.log('contentEncoding %s', contentEncoding);

			  	response.on('data', function (chunk) {
			    	//console.log('\nBODY: %s', chunk);
		            if( contentEncoding.match("gzip") ||
		                contentEncoding.match("decompress")){
		                //zipped content
		                binary = true;
						//Not supported yet
		                xhr.responseText += (chunk+'');
		            }else{
		                //this is a text file
		                xhr.responseText += (chunk+'');
		            }
					if(xhr.onreadystatechange){
						xhr.onreadystatechange();
					}
			  	});
				if(xhr.onreadystatechange){
					xhr.onreadystatechange();
				}
			});
	    }
        //write data to output stream if required
		//TODO: if they have set the request header for a chunked
		//request body, implement a chunked output stream

        //console.log('sending request %s\n', xhr.url);
        if(data){
            if(data instanceof Document){
                if ( xhr.method == "PUT" || xhr.method == "POST" ) {
                    xml = (new XMLSerializer()).serializeToString(data);
					request.write(xml);
                }
            }else if(data.length&&data.length>0){
                if ( xhr.method == "PUT" || xhr.method == "POST" ) {
                    request.write(data);
                }
            }
            request.end();
        }else{
            request.end();
        }
    }

};

/**
 * @author john resig & the envjs team
 * @uri http://www.envjs.com/
 * @copyright 2008-2010
 * @license MIT
 */
//CLOSURE_END
}());
