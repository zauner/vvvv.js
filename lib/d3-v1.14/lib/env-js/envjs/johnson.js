// This file must be executed automatically on startup for ruby/v8 rubyracer support.
var __this__ = this;

var require = (function() {
    var cached = {};
    var currentPath = Ruby.ENV['PWD'];
    var paths = [currentPath];
    
    function normalize(id) {
		var file;
        id = id + '.js';
        if (/^\.\.?/.test(id)) {
            // relative path
			try{ return Ruby.File.open(id, 'r') }catch(e){}
        } else {
            for (var i = 0, len = paths.length; i < len; ++i) {
			    try{ return Ruby.File.open(paths[i]+'/'+id, 'r') }catch(e){}
            }
        }
        return undefined;
    };
    
    
    function require(id) {
		//print('require :'+ id);
        var file = normalize(id);
        if (!file) {
            throw new Error("couldn't find module \"" + id + "\"");
        }
        if (!cached.hasOwnProperty(id)) {
			//print('loading '+id);
            var source = file.read();
            source = source.replace(/^\#\!.*/, '');
            source = "(function (require, exports, module) { "+source+"\n});";
            cached[id] = {
                exports: {},
                module: {
                    id: id,
                    uri: id
                }
            };
            var previousPath = currentPath;
            try {
                currentPath = id.substr(0, id.lastIndexOf('/')) || '.';
                var func = global.evaluate( source, id );
                func(require, cached[id].exports, cached[id].module);
            } finally {
                currentPath = previousPath;
            }
        }
		/*
		print('returning exports for id: '+id+' '+cached[id].exports);
		for(var prop in cached[id].exports){
			print('export: '+prop);
		}
		*/
        return cached[id].exports;
    };
    
    require.paths = paths;
    
    return require;
})();

require('./platform/johnson');
require('./window');
