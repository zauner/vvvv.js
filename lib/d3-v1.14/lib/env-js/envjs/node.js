process.env.TZ = 'America/Los_Angeles';

require('./platform/node');
Envjs.eventLoop = function() {}; // disabled for clean shutdown

require('./window');
new Window(__this__); // initialize a blank window
