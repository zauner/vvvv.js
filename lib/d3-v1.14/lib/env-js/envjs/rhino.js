// This file is compiled into the jar and executed automatically on startup.
var __this__ = this;
var require = (function() {
    var cached = {};
    var currentPath = java.lang.System.getProperty('user.dir');
    var paths = [currentPath];
    
    function normalize(id) {
		var file;
        id = id + '.js';
        if (/^\.\.?/.test(id)) {
            // relative path
            file = new java.io.File(currentPath, id);
            if (file.isFile()) {
                return file.toURL();
            }
        } else {
            for (var i = 0, len = paths.length; i < len; ++i) {
                file = new java.io.File(paths[i], id);
                if (file.isFile()) {
                    return file.toURL();
                }
            }
            // try to get it from the jar as a last resort
            /*var url = rhoop.getClass().getResource('/' + id);
            if (url !== null) {
                return url;
            }*/
        }
        return undefined;
    };
    
    function read(connection) {
        var stream = connection.getInputStream();
        var bytes = java.lang.reflect.Array.newInstance(
                java.lang.Byte.TYPE, 4096);
        var bytesStream = new java.io.ByteArrayOutputStream();
        var bytesRead;
        while ((bytesRead = stream.read(bytes)) >= 0) {
            if (bytesRead > 0) {
                bytesStream.write(bytes, 0, bytesRead);
            }
        }
        return String(bytesStream.toString());
    };
    
    function require(id) {
		//print('require :'+ id);
        var url = normalize(id);
        if (!url) {
            throw new Error("couldn't find module \"" + id + "\"");
        }
        id = String(url.toString());
        if (!cached.hasOwnProperty(id)) {
            var source = read(url.openConnection());
            source = source.replace(/^\#\!.*/, '');
            source = (
                "(function (require, exports, module) { " + source + "\n});");
            cached[id] = {
                exports: {},
                module: {
                    id: id,
                    uri: id
                }
            };
            var previousPath = currentPath;
            try {
                currentPath = id.substr(0, id.lastIndexOf('/')) || '.';
                var ctx = org.mozilla.javascript.Context.getCurrentContext();
                var func = ctx.evaluateString({}, source, id, 1, null);
                func(require, cached[id].exports, cached[id].module);
            } finally {
                currentPath = previousPath;
            }
        }
		/*
		print('returning exports for id: '+id+' '+cached[id].exports);
		for(var prop in cached[id].exports){
			print('export: '+prop);
		}
		*/
        return cached[id].exports;
    };
    
    require.paths = paths;
    
    return require;
}());
var __argv__ = arguments;
require('./platform/rhino');
require('./window');
