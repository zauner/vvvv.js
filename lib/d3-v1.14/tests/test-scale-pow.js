require("./../lib/env-js/envjs/node");
require("./../d3");

var f = d3.format(" .3f");

var x = d3.scale.pow().exponent(2);
console.log("domain([0, 1]).range([0, 1]):");
console.log("         -0.5  -> ", f(x(-0.5)));
console.log("          0.0  -> ", f(x(0.0)));
console.log("          0.5  -> ", f(x(0.5)));
console.log("          1.0  -> ", f(x(1.0)));
console.log("          1.5  -> ", f(x(1.5)));
console.log("");

var x = d3.scale.pow().exponent(2).domain([1, 2]);
console.log("domain([1, 2]).range([0, 1]):");
console.log("          0.5  -> ", f(x(0.5)));
console.log("          1.0  -> ", f(x(1.0)));
console.log("          1.5  -> ", f(x(1.5)));
console.log("          2.0  -> ", f(x(2.0)));
console.log("          2.5  -> ", f(x(2.5)));
console.log("");

var x = d3.scale.pow().exponent(2).domain([new Date(1990, 0, 1), new Date(1991, 0, 1)]);
console.log("domain([01/01/1990, 01/01/1991]).range([0, 1]):");
console.log("   10/20/1989  -> ", f(x(new Date(1989, 09, 20))));
console.log("   01/01/1990  -> ", f(x(new Date(1990, 00, 01))));
console.log("   03/15/1990  -> ", f(x(new Date(1990, 02, 15))));
console.log("   05/27/1990  -> ", f(x(new Date(1990, 04, 27))));
console.log("   01/01/1991  -> ", f(x(new Date(1991, 00, 01))));
console.log("   03/15/1991  -> ", f(x(new Date(1991, 02, 15))));
console.log("");

var x = d3.scale.pow().exponent(2).range(["red", "blue"]);
console.log("domain([0, 1]).range([\"red\", \"blue\"]):");
console.log("         -0.5  -> ", x(-0.5));
console.log("          0.0  -> ", x(0.0));
console.log("          0.5  -> ", x(0.5));
console.log("          1.0  -> ", x(1.0));
console.log("          1.5  -> ", x(1.5));
console.log("");

var x = d3.scale.pow().exponent(2).range(["red", "blue"]).interpolate(d3.interpolateHsl);
console.log("domain([0, 1]).range([\"red\", \"blue\"]).interpolate(hsl):");
console.log("        -0.50  -> ", x(-0.50));
console.log("         0.00  -> ", x(0.00));
console.log("         0.50  -> ", x(0.50));
console.log("        âˆš0.50  -> ", x(Math.SQRT1_2));
console.log("         1.00  -> ", x(1.00));
console.log("         2.00  -> ", x(2.00));
console.log("");

var x = d3.scale.pow().exponent(2).domain([1, 2]).range([0, 1]).clamp(true);
console.log("domain clamping:")
console.log("        under  -> ", x(-1));
console.log("         over  -> ", x(2));
console.log("");
