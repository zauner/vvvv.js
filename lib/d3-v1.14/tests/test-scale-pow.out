domain([0, 1]).range([0, 1]):
         -0.5  ->   0.250
          0.0  ->   0.000
          0.5  ->   0.250
          1.0  ->   1.000
          1.5  ->   2.250

domain([1, 2]).range([0, 1]):
          0.5  ->  −0.250
          1.0  ->   0.000
          1.5  ->   0.417
          2.0  ->   1.000
          2.5  ->   1.750

domain([01/01/1990, 01/01/1991]).range([0, 1]):
   10/20/1989  ->  −0.194
   01/01/1990  ->   0.000
   03/15/1990  ->   0.196
   05/27/1990  ->   0.394
   01/01/1991  ->   1.000
   03/15/1991  ->   1.206

domain([0, 1]).range(["red", "blue"]):
         -0.5  ->  rgb(191,0,64)
          0.0  ->  rgb(255,0,0)
          0.5  ->  rgb(191,0,64)
          1.0  ->  rgb(0,0,255)
          1.5  ->  rgb(-319,0,574)

domain([0, 1]).range(["red", "blue"]).interpolate(hsl):
        -0.50  ->  #ffff00
         0.00  ->  #ff0000
         0.50  ->  #ffff00
        √0.50  ->  #00ff00
         1.00  ->  #0000ff
         2.00  ->  #0000ff

domain clamping:
        under  ->  0
         over  ->  1

